src/Application/Auth/SalesforceAuthMiddleware.php

```diff
@@ -36,11 +37,25 @@ public function __construct(
     #[\Override]
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        if (!$this->verify($request)) {
+        $givenHash = $request->getHeaderLine('x-send-verify-hash');
+        $content = $request->getBody()->getContents();
+
+        if (!$this->verify($givenHash, $content)) {
             return $this->unauthorised($this->logger);
         }
 
-        return $handler->handle($request);
+        // because we've consumed the body content we now have to put the content back into a new request
+        // body so that the next handler can read it. More awkward than I'd like, but followed method from
+        // https://evertpot.com/222/
+        //
+        // This wasn't an issue up till now because all the requests from SF happened to have empty bodies.
+
+        $stream = fopen('php://memory', 'rb+');
+        \assert(\is_resource($stream));
+        fwrite($stream, $content);
+        rewind($stream);
+
+        return $handler->handle($request->withBody(new Stream($stream)));
```

Comment:
took me a while to work this out. Second time I've been bitten by `$request->getBody()->getContents()` not just taking a copy of the content but removing them from the stream.

Anyway good that I wrote a test that covered this instead of waiting to discover the issue when we actually try to integrate with SF.